File: ecommerc_dotnet/dto/Response/ProductResponseDto.cs
using ecommerc_dotnet.Validation;

namespace ecommerc_dotnet.dto.Request;

public class ProductResponseDto
{
    public Guid id { get; set; }
    [StringNotNullOrEmptyValidation(ErrorMessage= "name must not be empty")]
    public string name { get; set; }
    [StringNotNullOrEmptyValidation(ErrorMessage= "description must not be empty")]
    public string description { get; set; }
    public string  thmbnail { get; set; }
    public Guid subcategoryId { get; set; }
    public Guid categoryId { get; set; }
    public Guid storeId { get; set; }
    public decimal price { get; set; }
    public List<List<ProductVarientResponseDto>>? productVarients { get; set; }
    public List<string> productImages { get; set; }
}

File: ecommerce-delivery-man/app/src/main/java/com/example/e_commerc_delivery_man/dto/response/ProductResponseDto.kt
package com.example.e_commerc_delivery_man.dto.response

import androidx.compose.runtime.Composable
import com.example.hotel_mobile.services.kSerializeChanger.UUIDKserialize
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class ProductResponseDto(
    @Serializable(with= UUIDKserialize::class)
    val id: UUID,
    val name: String,
    val description:String,
    val thmbnail:String,
    @Serializable(with = UUIDKserialize::class)
    val subcategory_id: UUID,
    @Serializable(with = UUIDKserialize::class)
    val store_id: UUID,
    @Serializable(with = UUIDKserialize::class)
    val category_id: UUID,
    val price: Double,
    val productVarients:List<List<ProductVarientReponseDto>>?=null,
    val productImages:List<String>

) ;